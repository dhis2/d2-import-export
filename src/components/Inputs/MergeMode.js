import React from 'react'
import i18n from '@dhis2/d2-i18n'
import { mergeModeOptions } from '../../utils/options'
import { RadioGroupField } from '../'

const NAME = 'mergeMode'
const DATATEST = 'input-merge-mode'
const LABEL = i18n.t('Merge mode')

const MergeMode = () => (
    <RadioGroupField
        name={NAME}
        label={LABEL}
        options={mergeModeOptions}
        dataTest={DATATEST}
    />
)

export { MergeMode }
