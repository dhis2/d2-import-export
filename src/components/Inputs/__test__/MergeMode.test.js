import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { MergeMode } from '../MergeMode'

jest.mock('react-final-form')

describe('Input component - MergeMode', () => {
    it('should render correctly', () => {
        const file = shallow(<MergeMode />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
