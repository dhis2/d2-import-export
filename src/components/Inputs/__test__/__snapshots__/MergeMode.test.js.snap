// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - MergeMode should render correctly 1`] = `
<RadioGroup
  dataTest="input-merge-mode"
  label="Merge mode"
  name="mergeMode"
  options={
    Array [
      Object {
        "label": "Merge",
        "value": "MERGE",
      },
      Object {
        "label": "Replace",
        "value": "REPLACE",
      },
    ]
  }
/>
`;
